README.md

```diff
@@ -1,20 +1,506 @@
+
+```js
+const connections = {};
+```
+
+This isn't too useful yet. Let's say we want to "bootstrap" our server with a
+couple of client connections or want to reserve a couple of hard-coded
+usernames:
+
+```js
+const connections = {
+    max:    conn0,
+    alex:   conn1,
+    james:  conn2
+};
+```
+
+Doesn't look too different from the above table, does it? Keys are on the left,
+values on the right.
+
+Since object keys are **always** "stringified" (meaning that keys are always
+strings, never numbers or anything else) in JavaScript, we can leave out the
+enclosing `'` or `"`. Nevertheless, leaving them in also works without any
+problems:
+
+```js
+const connections = {
+    "max":  conn0, // enclosing "
+    'alex': conn1, // enclosing '
+    james:  conn2
+};
+```
+
+An interesting scenario is when you want to be able to initialize an object
+literal with something that hasn't been evaluated yet, e.g. let's say you want
+to create an object with the current date as a key. Thankfully ES6 defines an
+new syntax to achieve this:
+
+```js
+{ [Date.now()]: null }; // #=> { '1469397041282': null }
+```
+
+Keep in mind that keys are **always** strings, never arrays. While in the above
+example we enclose the key in brackets, this does not indicate that the key
+*itself* is an array.
+
+### Constructor functions
+While object keys are strings, object values can be anything. This is especially
+handy when using constructor functions. Constructor functions allow you to
+create objects using a particular "template" (called a "prototype"). They are
+commonly used (especially pre-ES6) for implementing class-systems.
+
+For example, in a "traditional" object-oriented language (such as Java), you
+might create a class "Server" to encapsulate the server functionality. Turns out
+we can do the same thing in JavaScript as well!
+
+```js
+function Server () {
+    this.createdAt = new Date();
+}
+const server = new Server();
+server.createdAt; // #=> Sun Jul 24 2016 12:59:51 GMT+0100 (BST)
+```
+
+`new Server()` creates a new object of class `Server`. This isn't too useful
+yet, but let's look at what happens when we define some methods:
+
+```js
+function Server () {}
+Server.prototype.start = function () {
+    return 'not implemented';
+}
+Server.prototype.stop = function () {
+    return 'not implemented';
+}
+const server = new Server();
+server.start(); // #=> "not implemented"
+```
+
+But where does the `prototype` come from now? The `prototype` property is kind
+of special. It indicates that all objects created using the `Server` constructor
+"inherit" ("true" inheritance comes later) those properties defined on the
+`prototype`. In this specific case, this means that all servers will have a
+`start` method.
+
+**Advanced:** In ES6 we finally got the "class" syntax, but it's mostly just
+syntactic sugar around the above constructor functions.
+
+### Object.create()
+One problem arising from the use of prototypes in JavaScript is that accessing
```

Comment:
Students might also find this section a bit challenging, but it's super thorough -- great job.